chia/full_node/full_node_api.py

```diff
@@ -1314,13 +1314,34 @@ async def request_puzzle_solution(self, request: wallet_protocol.RequestPuzzleSo
         return response_msg
 
     @api_request
-    async def request_header_blocks(self, request: wallet_protocol.RequestHeaderBlocks) -> Optional[Message]:
+    async def request_header_block_blobs(self, request: wallet_protocol.RequestHeaderBlockBlobs) -> Optional[Message]:
         if request.end_height < request.start_height or request.end_height - request.start_height > 32:
             return None
+        height_to_hash = self.full_node.blockchain.height_to_hash
+        header_hashes: List[bytes32] = []
+        for i in range(request.start_height, request.end_height + 1):
+            header_hash: Optional[bytes32] = height_to_hash(uint32(i))
+            if header_hash is None:
+                reject = RejectHeaderBlocks(request.start_height, request.end_height)
+                msg = make_msg(ProtocolMessageTypes.reject_header_blocks, reject)
+                return msg
+            header_hashes.append(header_hash)
 
+        blocks: List[bytes] = await self.full_node.block_store.get_block_blobs_by_hash(header_hashes)
```

Comment:
We seem to be returning full blocks, but we need header blocks right?

Reply:
We are returning a large amount of data, and forcing the wallet to parse it all which is very slow